sparkplug-core/src/java/sparkplug/function/SerializableFn.java

```diff
@@ -5,56 +5,146 @@
 import clojure.lang.IFn;
 import clojure.lang.RT;
 import clojure.lang.Symbol;
+import clojure.lang.Var;
 
 import java.io.IOException;
 import java.io.InvalidObjectException;
+
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 import java.io.Serializable;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 
 /**
  * Base class for function classes built for interop with Spark and Scala.
+ *
+ * This class is designed to be serialized across computation boundaries in a
+ * manner compatible with Spark and Kryo, while ensuring that required code is
+ * loaded upon deserialization.
  */
 public abstract class SerializableFn implements Serializable {
 
+    private static final Logger logger = LoggerFactory.getLogger(SerializableFn.class);
+    private static final Var require = RT.var("clojure.core", "require");
+
     protected IFn f;
+    protected List<String> namespaces;
 
 
     /**
      * Default empty constructor.
+     *
+     * TODO is this accurate?
+     * Inheriting classes must also have an empty constructor to be properly
+     * serializable.
      */
     private SerializableFn() {
     }
 
 
-    private void writeObject(ObjectOutputStream out) throws IOException {
+    /**
+     * Construct a new serializable wrapper for the function with an explicit
+     * set of required namespaces.
+     *
+     * @param fn Clojure function to wrap
+     * @param namespaces collection of namespaces required
+     */
+    protected SerializableFn(IFn fn, Collection<String> namespaces) {
+        this.f = fn;
+        this.namespaces = new ArrayList<String>(namespaces);
```

Comment:
A very minor best-practice you could add here would be to do the sorting against a local variable and then move this assignment to the end with a `Collections.unmodifiableList(...)` wrapping it.